analytical/templatetags/matomo.py

```diff
@@ -97,6 +97,19 @@ def render(self, context):
         if getattr(settings, 'MATOMO_DISABLE_COOKIES', False):
             commands.append(DISABLE_COOKIES_CODE)
 
+        if getattr(settings, "MATOMO_REQUIRE_CONSENT", False):
```

Comment:
Add something to check if registered users are to be tracked no matter what (since they normally have to accept TOS and such). Someone may have another opinion on the best option but I'm going to require a setting called `MATOMO_TRACK_REGISTERED` which, if true (and the user is logged in), will **not** push the requireConsent command to `_paq`

Reply:
**for anyone who sees this**: What do you think about making a very simple context provider that checks if the user is logged in, if so don't render the consent code if  `settings.ALWAYS_TRACK_REGISTERED` is `True`. Otherwise if the user is not logged in or that setting is set to False (or not set) pass the script to the template.
Then in mamoto.py in the template tag, make it render something to actually inject the script by adding something like this to the end of whatever scripts are returned. `{{consent_script}}`
If we ALWAYS have that part at the end it should always render correctly.

This is the only way I could come up with